### PR TITLE
Used Parsec library to parse department names and codes.

### PR DESCRIPTION
**Pull request by alexbaluta**

### CODE REVIEW

File: hs/WebParsing/TimeTableParser.hs [hunk 1]
```diff
@@ -36,38 +39,32 @@ insertDepartment :: MonadIO m => ([T.Text], T.Text) -> ReaderT SqlBackend m ()
...
+splitTagsToLists :: String -> Either ParseError [[String]]
+splitTagsToLists input =
+    let tags = endBy line eol
+        line = sepBy cell (char '[')
+        cell = many (noneOf ",\n")
+        eol = char '\n'
+    in  parse tags "(unknown)" input
+
+-- | Extracts tuples containing the department names and respective codes from raw data.
+extractDeptInfo :: [[[String]]] -> [([T.Text],T.Text)]
```
Comment (Ian-Stewart-Binks) [inline hs/WebParsing/TimeTableParser.hs]: Add a space in the tuple.

**Comment by Ian-Stewart-Binks** [comment]
Nice!

File: hs/WebParsing/TimeTableParser.hs (hunk 1 above)
Comment (david-yz-liu) [inline hs/WebParsing/TimeTableParser.hs]: This is at least one nested list level too high. Its signature should be `[[String]] -> ([T.Text], T.Text)`, and you should `map` it below.

File: hs/WebParsing/TimeTableParser.hs [hunk 2]
```diff
@@ -36,38 +39,32 @@ insertDepartment :: MonadIO m => ([T.Text], T.Text) -> ReaderT SqlBackend m ()
...
-                "and",
-                "-",
-                "[",
-                "]",
-                ""]
+-- | Extracts a list of lists of department names followed by unfiltered department codes.
+splitTagsToLists :: String -> Either ParseError [[String]]
+splitTagsToLists input =
+    let tags = endBy line eol
+        line = sepBy cell (char '[')
```
Comment (david-yz-liu) [inline hs/WebParsing/TimeTableParser.hs]: @alexbaluta Okay, I thought about this some more and figured out a nicer way of doing it. Look into the `manyTill` and `oneOf` Parsec functions. You can also chain parsers by doing something like

```
(do
  x <- parser1
  y <- parser2
  return (x, y)
)

Comment (alexbaluta) [inline hs/WebParsing/TimeTableParser.hs]: Thanks! I've changed extractDeptInfo to match the type signature you mentioned and simplified the code. I'll look into the manyTill and oneOf functions. It's an interesting solution!

**Comment by david-yz-liu** [comment]
@alexbaluta Definitely on the right track. Keep going!

**Comment by david-yz-liu** [comment]
@alexbaluta I'm going to close this for now, but please keep working on this and keep me posted!

**Pull request closed without merging**